File: Linked List/Arrays/80. Remove Duplicates from Sorted Array II/fighting.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if not nums:
            return None

        count = 1
        place = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[place] = nums[i]
                place += 1
                count = 1
            else:
                if count == 1:
                    nums[place] = nums[i]
                    place += 1
                    count = 2
        return place

    def removeDuplicates2(self, nums: list[int]) -> int:
        if not nums:
            return None
        elif len(nums) < 2:
            return len(nums)
        
        place = 2
        for i in range(2, len(nums)):
            if nums[i] != nums[place - 2]:
                nums[place] = nums[i]
                place += 1
        return place

solution = Solution()
nums = [1,1,1,2,2,3]
print(solution.removeDuplicates(nums))
print(nums)

File: Linked List/Arrays/80. Remove Duplicates from Sorted Array II/fighting.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        const int FIXED_NUMBER = 2;
        if (nums.size() < FIXED_NUMBER)
        {
            return nums.size();
        }

        int place = FIXED_NUMBER;
        for (int i = FIXED_NUMBER; i < nums.size(); i++)
        {
            if (nums[i] != nums[place - FIXED_NUMBER])
            {
                nums[place] = nums[i];
                place++;
            }
        }
        return place;
    }
};

int main()
{
    vector<int> nums;
    nums.push_back(1);
    nums.push_back(1);
    nums.push_back(1);
    nums.push_back(2);
    nums.push_back(2);
    nums.push_back(2);
    nums.push_back(3);
    nums.push_back(4);

    Solution temp;
    int x = temp.removeDuplicates(nums);
    cout << x;
    cout << "\n";
    for (int i = 0; i < nums.size(); i++)
    {
        cout << nums[i];
        cout << " ";
    };
}
